source/compiler/sc5.c

```diff
@@ -220,13 +225,22 @@ static int errwarn;
  *                     fcurrent   (reffered to only)
  *                     errflag    (altered)
  */
-SC_FUNC int error(int number,...)
+SC_FUNC int error(long number,...)
 {
 static char *prefix[3]={ "error", "fatal error", "warning" };
 static int lastline,errorcount;
 static short lastfile;
   char *msg,*pre;
   va_list argptr;
+  char string[128];
+  int notice;
+
+  /* split the error field between the real error/warning number and an optional
+   * "notice" number
+   */
+  notice=(unsigned long)number >> (sizeof(long)*4);
+  number&=((unsigned long)~0) >> (sizeof(long)*4);
```

Comment:
Better as `~0UL >> sizeof (long) * 4`?

Reply:
I did it this way so there would be more similarity between this line and the above one.

Reply:
Except casting doesn't sign extend.  So this is:

`(unsigned long)0xFFFFFFFF`

if `sizeof (long) == sizeof (int)` that's fine, if it isn't, you will get `0x00000000FFFFFFFF`.  After shifting that will give 0, not `0xFFFF`.

Reply:
Oh, so that's what you meant. Yeah, I already noticed that mistake the first time you left a review comment at that line and fixed it locally by just moving the `~` before `(unsigned long)`. Thanks for pointing it out.